src/Raven.Client/Documents/Session/AsyncSessionDocumentTimeSeries.cs

```diff
@@ -332,16 +342,22 @@ private static TimeSeriesEntry[] MergeRangesWithResults(DateTime @from, DateTime
             return mergedValues.ToArray();
         }
 
-        private static IEnumerable<TimeSeriesEntry> ChopRelevantRange(TimeSeriesRangeResult range, DateTime from, DateTime to)
+        private static IEnumerable<TimeSeriesEntry> ChopRelevantRange(TimeSeriesRangeResult range, DateTime from, DateTime to, int start, int pageSize)
         {
             if (range.Entries == null)
                 yield break;
 
             foreach (var value in range.Entries)
             {
+                if (pageSize-- <= 0)
+                    yield break;
+
                 if (value.Timestamp > to)
                     yield break;
 
+                if (start-- > 0)
```

Comment:
start and pageSize should be done just before the return after all other filtering (start first, then pageSize)